azure-pipelines/azure-templates/docker-template.yml

```diff
@@ -1,33 +1,20 @@
 parameters:
-  - name: 'displayName'
-    type: string
-  - name: 'targetPath'
-    type: string
-  - name: 'artifact'
-    type: string
-  - name: 'tags'
+  - name: 'tag'
     type: string
   - name: 'Dockerfile'
     type: string
   - name: 'repository'
     type: string
+  - name: 'containerRegistry'
+    type: string
 
-  jobs:
-  - job: BuildPushAzure
-    displayName: ${{ parameters.displayName }}
-    steps:
-       
-    - task: DownloadPipelineArtifact@2
-      inputs:
-        buildType: 'current'
-        artifactName: ${{ parameters.artifact }}
-        itemPattern: '**.jar'
-        targetPath: ${{ parameters.targetPath }}
+steps:
 
-    - task: Docker@2
-      inputs:
-        containerRegistry: ${{ parameters.repository }}
-        repository: $(service)
-        command: 'buildAndPush'
-        Dockerfile: ${{ parameters.Dockerfile }}
-        tags: ${{ parameters.tags }}
\ No newline at end of file
+  - task: Docker@2
+    displayName: Create/Push Docker Image
+    inputs:
+      containerRegistry: ${{ parameters.containerRegistry }}
+      repository: ${{ parameters.repository }}
+      command: 'buildAndPush'
+      Dockerfile: ${{ parameters.Dockerfile }}
+      tags: ${{ parameters.tag }}
```

Comment:
a template as thin as this doesn't add much value over just using the step directly. i guess it does require these exact parameters rather than allowing you to forget to specify a tag or something. non-blocking issue.

Reply:
I agree with this, especially because of the nature of the pipeline. It dynamically creates the variables it needs during runtime, while templates need compile time variables--and it's just a single pipeline, so they don't add much. 

The use of the templates came down to it's only--small--value which is allowing you to change things like image tags etc... and as a project requirement's change having the template allows you to add more useful parameters and steps without restructuring the pipeline.